File: easy/Contains Duplicate I/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const map = {};
    for (let i = 0; i < nums.length; ++i) {
        if (map[nums[i]] != null) {
            return true;
        }
        map[nums[i]] = i;
    }
    return false;
};

const testContainsDuplicate = function(nums, expected) {
    const result = containsDuplicate(nums);
    if (result == expected) {
        console.log("Test Case Passed!");
    } else {
        console.log(
            `Test Case Failed: nums: ${nums} expected: ${expected} result: ${result}`
        );
    }
}

testContainsDuplicate([1,2,3,1], true);
testContainsDuplicate([1,2,3,4], false);
testContainsDuplicate([1,1,1,3,3,4,3,2,4,2], true);

File: second_pass/1_arrays_and_hashing/1_contains_duplicate/contains_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(list(set(nums))) != len(nums)

    def testContainsDuplicate(self, nums, expected) -> bool:
        actual = self.containsDuplicate(nums)
        
        if actual == expected:
            print ('Test Case Passed!')
        else:
            print (f'Test Case Failed! nums: {nums} expected: {expected} actual: {actual}')

solution = Solution()

solution.testContainsDuplicate([1,2,3,1], True)
solution.testContainsDuplicate([1,2,3,4], False)
solution.testContainsDuplicate([1,1,1,3,3,4,3,2,4,2], True)